### DISCUSSION TITLE
Improve documentation on how to test GitPython programs

### DISCUSSION THREAD

**Discussion by lyz-code** [gitpython-developers/GitPython#1205] [Q&A] [answered]
I've started developing a program that uses gitpython some weeks ago, and I had to spend a considerable amount of time deducing how to test it. I think it would be good to have a section in the official docs giving some insights.

I've published how I did it [here](https://lyz-code.github.io/blue-book/coding/python/gitpython/#testing). I don't know if it's the best approach, but if you like it, I can make a PR to include the content to the official docs.

**Comment by Byron** [comment, marked as answer]
Thanks for the suggestion, it can't hurt to provide a little more information on how to test with GitPython from within your own project.

Please feel free to submit a PR and we continue the conversation there.
